### PR TITLE
fix(query): enabled reactivity fix

### LINKED ISSUE #943: Fix Enabled Reactivity
## What is the issue?
We pass a reactive (ref/computed) object to the enabled field of the query object in `useSomething()`. The generated enabled field incorrectly evaluates to true even when the query param(s) are undefined leading to queries such as `updateId/undefined`.

## What is happening?
```JavaScript
const getSomethingQuery = useGetSomething(computed(() => something));

...

// generated code
export const getSomethingQueryOptions = (..) => {
    ...
    // 'something' is of type MaybeRef<string> (for example)
    return { queryKey, ..., enabled: !!something }
}
```

## What is expected?
```JavaScript
export const getSomethingQueryOptions = (..) => {
    ...
    return { queryKey, ..., enabled: computed(() => !!unref(something) }
}
```

### PR DESCRIPTION
**Pull request by BenGane**
## Status

**READY/WIP**

Fix #943 

## Description

Addresses the issue outlined in https://github.com/anymaniax/orval/issues/943

### CODE REVIEW

**Review by Maxim-Mazurok — APPROVED** [review]
Teammate here, this fix has been working well for us 👍

**Comment by anymaniax** [comment]
Thanks for this @BenGane. There is a difference with #931?

**Comment by Maxim-Mazurok** [comment]
> Thanks for this @BenGane. There is a difference with #931?

Yes, this one is making query `enabled` reactive, and #931 is making query parameters reactive. Not the same thing, and we use both patches currently.

I also wanted to give a little update that I've created a workitem for myself at work to go over all the issues PRs and make sure they are in a good shape so we can get them merged and get rid of patches that we use internally. It's in our quick&important project so it's likely I'll get to this pretty soon. Cheers!

**Comment by anymaniax** [comment]
oki thanks @Maxim-Mazurok!

**Review by anymaniax — APPROVED** [review]
Thanks @BenGane!

File: samples/vue-query/src/api/endpoints/petstoreFromFileSpecWithTransformer.ts [hunk 1]
```diff
@@ -307,7 +317,12 @@ export const getShowPetByIdInfiniteQueryOptions = <
     signal,
   }) => showPetById(petId, version, signal);
 
-  return { queryKey, queryFn, enabled: !!(version && petId), ...queryOptions };
+  return {
+    queryKey,
+    queryFn,
+    enabled: computed(() => !!unref(version && petId)),
```
Comment (WickyNilliams) [inline samples/vue-query/src/api/endpoints/petstoreFromFileSpecWithTransformer.ts:323]: is this correct? don't you want:

```ts
!!(unref(version) && unref(petId))
```

Comment (Maxim-Mazurok) [reply samples/vue-query/src/api/endpoints/petstoreFromFileSpecWithTransformer.ts:323]: Yes, great catch!

Verified like so:
```js
const x = computed(() => undefined);
const y = computed(() => 2);
const z1 = unref(x && y); // 2 <- unexpected
const z2 = unref(x) && unref(y); // undefined <- expected
```

**Review by Maxim-Mazurok — CHANGES_REQUESTED** [review]

File: samples/vue-query/src/api/endpoints/petstoreFromFileSpecWithTransformer.ts (hunk 1 above)
Comment (anymaniax) [reply samples/vue-query/src/api/endpoints/petstoreFromFileSpecWithTransformer.ts:323]: Is it ready to merged?

**Comment by melloware** [comment]
Looks like conflicts need to be resolved...

**Comment by melloware** [comment]
@BenGane @Maxim-Mazurok where are we with this PR?

**Comment by Maxim-Mazurok** [comment]
Hello, as I've mentioned in other issues I've been busy with other work, however the Orval PRs are next on my list, I'd expect to see some progress next week. Cheers!

**Pull request closed without merging**